src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs

```diff
@@ -11999,5 +11999,161 @@ public static void F()
             await VerifyItemIsAbsentAsync(source, "goo");
             await VerifyItemIsAbsentAsync(source, "Bar");
         }
+
+        [Fact]
+        public async Task ParameterAvailableInMethodAttributeNameof()
+        {
+            var source = @"
+class C
+{
+    [Some(nameof(p$$))]
+    void M(int parameter) { }
+}
+";
+            await VerifyItemExistsAsync(MakeMarkup(source), "parameter");
+
+            await VerifyItemIsAbsentAsync(MakeMarkup(source, languageVersion: "10"), "parameter");
+        }
+
+        [Fact]
+        public async Task ParameterNotAvailableInMethodAttributeNameofWithNoArgument()
+        {
+            var source = @"
+class C
+{
+    [Some(nameof($$))]
+    void M(int parameter) { }
+}
+";
+            // An invocation may not be considered a 'nameof' operator unless it has one argument
+            await VerifyItemIsAbsentAsync(MakeMarkup(source), "parameter");
```

Comment:
I think it would be good to support this. It can be very helpful for `CallerArgumentExpression` too:

```csharp
void M(int arg, [CallerArgumentExpression(nameof($$))] int parameter) { }
```

It would be good to suggest `arg` without having to type `a`

Reply:
Agreed, but that will likely have to be left for IDE layer to refine, as compiler API is behaving by-design. I think there's already IDE logic to handle `nameof` specially, and it would have to be expanded in some later PR.